clang/lib/Sema/SemaInit.cpp

```diff
@@ -4576,7 +4576,9 @@ static void TryConstructorInitialization(Sema &S,
     if (!IsListInit &&
```

Comment:
nit: I think this if condition is becoming complex enough that it's worth adding a comment for explanation.